pkg/ebpf/c/tracee.bpf.c

```diff
@@ -6289,3 +6313,836 @@ int tc_ingress(struct __sk_buff *skb)
+// flow does not have preemption disabled, so the map used in between the 3
+// hooks need to use something that is available to all 3 of them.
+
+// At the end, the logic is simple: every time a socket is created an inode is
+// also created. The task owning the socket is indexed by the socket inode so
+// everytime this socket is used we know which task it belongs to (specially
+// during ingress hook).
+
+//
+// network helper functions
+//
+
+static __always_inline bool is_socket_supported(struct socket *sock)
+{
+    // clang-format off
+    struct sock *sk = (void *) BPF_READ(sock, sk);
+
+    u8 family = BPF_READ(sk, __sk_common.skc_family);
+    switch (family) {
+        // case PF_IB:
+        case PF_INET:
+        case PF_INET6:
+            break;
+        default:
+            return 0;
+    }
+    // TODO: kernel 5.4 checks/problems (wrong relocation ?)
+    // u16 type = BPF_CORE_READ_BITFIELD_PROBED(sk, sk_type);
+    // switch (type) {
+    //     case SOCK_STREAM:
+    //     case SOCK_DGRAM:
+    //         break;
+    //     default:
+    //         return 0;
+    // }
+    u16 protocol = get_sock_protocol(sk);
+    switch (protocol) {
+        // case IPPROTO_IPV6:
+        // case IPPROTO_IPIP:
+        // case IPPROTO_DCCP:
+        // case IPPROTO_SCTP:
+        // case IPPROTO_UDPLITE:
+        case IPPROTO_IP:
+        case IPPROTO_TCP:
+        case IPPROTO_UDP:
+        case IPPROTO_ICMP:
+        case IPPROTO_ICMPV6:
+            break;
+        default:
+            return 0;
+    }
+
+    return 1;
+    // clang-format on
+}
+
+//
+// network related maps
+//
+
+// clang-format off
+
+// cgroupctxmap
+
+#define SUB_NET_PACKET_IP      (1 << 0) // submit net_packet_ip_base
+#define SUB_NET_PACKET_TCP     (1 << 1) // submit net_packet_tcp_base
+#define SUB_NET_PACKET_UDP     (1 << 2) // submit net_packet_udp_base
+#define SUB_NET_PACKET_ICMP    (1 << 3) // submit net_packet_icmp_base
+#define SUB_NET_PACKET_ICMPV6  (1 << 4) // submit net_packet_icmpv6_base
+#define SUB_NET_PACKET_DNS     (1 << 5) // submit net_packet_dns_base
+
+typedef struct net_event_context {
+    event_context_t eventctx;
+    struct { // event arguments (needs packing), use anonymous struct to ...
+        u8 index0;
+        u32 bytes;
+        // ... (payload sent by bpf_perf_event_output)
+    } __attribute__((__packed__)); // ... avoid address-of-packed-member warns
+    // members bellow this point are metadata (not part of event to be sent)
+    u8 should_submit;
+    u32 header_size;
+} net_event_context_t;
+
+struct {
+    __uint(type, BPF_MAP_TYPE_HASH);
+    __uint(max_entries, 1024);          // simultaneous cgroup/skb ingress/eggress progs
+    __type(key, u64);                   // sk_buff timestamp
+    __type(value, net_event_context_t); // event context built so cgroup/skb can use
+} cgrpctxmap SEC(".maps");              // saved info between SKB caller and SKB program
```

Comment:
i think its cleaner to define the network maps in tracee.bpf.c maps part, with the macros tracee uses

Reply:
Ditto (on using helpers for doing something that libbpf already has a standard for). Also, based on Yaniv's suggestion, I'm spinning the new network code off the BPF source code. Let's see how that goes, and we can revisit using or not helpers.